src/main/webapp/app/entities/exercise/exercise-utils.ts

```diff
@@ -108,3 +111,22 @@ const participationStatusForModelingTextFileUploadExercise = (exercise: Exercise
         return ParticipationStatus.UNINITIALIZED;
     }
 };
+
+/**
+ * Checks whether the given exercise is eligible for receiving manual results.
+ * This is the case if the user is at least a tutor and the exercise itself is a regular exercise, or a programming
+ * exercise for which manual reviews have been enabled. For programming exercises, the due date also has to be in the
+ * past.
+ *
+ * @param exercise
+ */
+export const areManualResultsAllowed = (exercise: Exercise) => {
+    if (exercise.type === ExerciseType.PROGRAMMING) {
+        // Only allow new results if manual reviews are activated and the due date/after due date has passed
+        const exc = exercise as ProgrammingExercise;
+        const relevantDueDate = exc.buildAndTestStudentSubmissionsAfterDueDate ? exc.buildAndTestStudentSubmissionsAfterDueDate : exc.dueDate;
+        return exc.isAtLeastTutor && exc.assessmentType === AssessmentType.SEMI_AUTOMATIC && moment(relevantDueDate!).isBefore(now());
```

Comment:
what happens if relevantDueDate is null or undefined?

Reply:
I added a null check, so that manual reviews are allowed if the relevant date is null

Reply:
same problem as above in the Java code. `relevantDueDate` could be null or undefined if both `exc.buildAndTestStudentSubmissionsAfterDueDate` and `exc.dueDate` are null or undefined. In this case, a TypeScript error would occur on runtime, because you use `!` in line 128.

Please add a proper null check after line 127 and before 128. In case `relevantDueDate` is null or undefined, manual results are not possible